include/types/announcement.class.php

```diff
@@ -175,17 +176,15 @@ function validate() {
 
         // attribute type is optional
         if (!is_null($this->type)) {
-            $allowedValues =  array("WARNING", "ERROR", "INFORMATION", "SYSTEM");
+            $allowedValues =  array("INFORMATION", "SYSTEM");
             if ($this->isString($this->type, 'type', $allowedValues) === false)
                 return false;
         }
 
-        // attribute priority is optional
-        if (!is_null($this->priority)) {
-            if ($this->isInteger($this->priority, 'priority', 3, 1) === false)
-                return false;
-        }
-
+        // attribute priority is required
+        if (in_array($this->priority, array("HIGH","MEDIUM","LOW")) === false)
```

Comment:
Although ```in_array``` works, no error message is set so we can't log the reason why validation failed.

Use ```$this->isString($this->priority, 'priority', $allowedValues)``` method instead, which takes care of setting the error message.